src/test/java/org/gridsuite/study/server/NetworkModificationUnitTest.java

```diff
@@ -0,0 +1,197 @@
+    @Autowired
+    NodeRepository nodeRepository;
+    @Autowired
+    NetworkModificationNodeInfoRepository networkModificationNodeInfoRepository;
+    @Autowired
+    NetworkModificationTreeService networkModificationTreeService;
+    @Autowired
+    StudyRepository studyRepository;
+    @Autowired
+    StudyController studyController;
+    @MockBean
+    ReportService reportService;
+    @MockBean
+    NetworkStoreService networkStoreService;
+    @MockBean
+    NetworkService networkService;
+    @Value("${loadflow.default-provider}")
+    String defaultLoadflowProvider;
+
+    private static final String CASE_LOADFLOW_UUID_STRING = "11a91c11-2c2d-83bb-b45f-20b83e4ef00c";
+    private static final UUID CASE_LOADFLOW_UUID = UUID.fromString(CASE_LOADFLOW_UUID_STRING);
+    private static final UUID NETWORK_UUID = UUID.fromString("38400000-8cf0-11bd-b23e-10b96e4ef00d");
+    private static final long TIMEOUT = 1000;
+    private static final String VARIANT_1 = "variant_1";
+
+    private UUID studyUuid;
+    private UUID node1Uuid;
+    private UUID node2Uuid;
+    private UUID node3Uuid;
+
+    //output destinations
+    @Autowired
+    private OutputDestination output;
+    private final String studyUpdateDestination = "study.update";
+
+    @BeforeEach
+    public void setup() {
+        StudyEntity study = insertStudy(NETWORK_UUID, CASE_LOADFLOW_UUID);
+        studyUuid = study.getId();
+
+        NodeEntity rootNode = nodeRepository.save(new NodeEntity(null, null, NodeType.ROOT, study, false, null));
+
+        NodeEntity node1 = insertNode(study, node1Uuid, rootNode, BuildStatus.BUILT);
+        NodeEntity node2 = insertNode(study, node2Uuid, node1, BuildStatus.BUILT);
+        NodeEntity node3 = insertNode(study, node3Uuid, node1, BuildStatus.NOT_BUILT);
+
+        node1Uuid = node1.getIdNode();
+        node2Uuid = node2.getIdNode();
+        node3Uuid = node3.getIdNode();
+    }
+
+    @Test
+    void unbuildNode() {
+        /**
+         *       rootNode
+         *          |
+         *       node1(B)
+         *     |         |
+         *  node2(B)   node3
+         */
+        List<NetworkModificationNodeInfoEntity> nodesInfos = networkModificationNodeInfoRepository.findAll();
+        NetworkModificationNodeInfoEntity node1Infos = nodesInfos.stream().filter(n -> n.getIdNode().equals(node1Uuid)).findAny().orElse(null);
+        NetworkModificationNodeInfoEntity node2Infos = nodesInfos.stream().filter(n -> n.getIdNode().equals(node2Uuid)).findAny().orElse(null);
+        NetworkModificationNodeInfoEntity node3Infos = nodesInfos.stream().filter(n -> n.getIdNode().equals(node3Uuid)).findAny().orElse(null);
+
+        assertEquals(BuildStatus.BUILT, node1Infos.getNodeBuildStatus().getLocalBuildStatus());
+        assertEquals(BuildStatus.BUILT, node2Infos.getNodeBuildStatus().getLocalBuildStatus());
+        assertEquals(BuildStatus.NOT_BUILT, node3Infos.getNodeBuildStatus().getLocalBuildStatus());
```

Comment:
It would be a bit cleaner to use orElseThrow here instead of letting it get a nullPointer after in the case the nodes are not found no ?